src/mailmime_write_generic.rs

```diff
@@ -1840,17 +1824,15 @@ pub unsafe fn mailmime_quoted_printable_write_driver(
                             return r;
                         }
                         start = text.offset(i as isize).offset(1isize);
-                        snprintf(
-                            hexstr.as_mut_ptr(),
-                            6i32 as libc::size_t,
-                            b"=%02X\x00" as *const u8 as *const libc::c_char,
-                            ch as libc::c_int,
-                        );
+                        let raw =
+                            format!("={}", hex::encode_upper(&(ch as libc::c_int).to_be_bytes()));
```

Comment:
Not `"={:02}`?

Reply:
the hex library will always encode an even number, and `to_be_bytes()` always returns a slice of bytes, even when 0. so that should not be needed

Reply:
switched it that, makes much more sense